src/Common/src/Interop/Unix/libc/Interop.getdomainname.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+internal static partial class Interop
+{
+    internal static partial class libc
+    {
+        [DllImport(Libraries.Libc, SetLastError = true)]
+        private static extern unsafe int getdomainname(byte* name, int len);
+
+        internal static unsafe string getdomainname()
+        {
+            const int HOST_NAME_MAX = 255; // man getdomainname
+            const int ArrLength = HOST_NAME_MAX + 1;
+
+            byte* name = stackalloc byte[ArrLength];
+            int errno = getdomainname(name, ArrLength);
+            if (errno == 0)
+            {
+                int nullPos = 0;
+                for (; nullPos < ArrLength && name[nullPos] != '\0'; nullPos++) ;
+                return Encoding.UTF8.GetString(name, nullPos);
```

Comment:
You should be able to replace the whole contents of this ```if``` block with ```return Marshal.PtrToStringAnsi((IntPtr)name);```, as that's UTF8 on Unix and already looks for the null terminator.